app/lib/dartdoc/dartdoc_runner.dart

```diff
@@ -40,6 +40,56 @@ class DartdocJobProcessor extends JobProcessor {
   DartdocJobProcessor({Duration lockDuration})
       : super(service: JobService.dartdoc, lockDuration: lockDuration);
 
+  /// Uses the tool environment's SDK (the one that is used for analysis too) to
+  /// generate dartdoc documentation and extracted data file for SDK API indexing.
+  /// Only the extracted data file will be used and uploaded.
+  Future generateDocsForSdk() async {
+    if (await dartdocBackend.hasValidDartSdkDartdocData()) return;
+    final tempDir =
+        await Directory.systemTemp.createTemp('pub-dartlang-dartdoc');
+    try {
+      final tempDirPath = tempDir.resolveSymbolicLinksSync();
+      final outputDir = tempDirPath;
+      final args = [
+        '--sdk-docs',
+        '--output',
+        outputDir,
+        '--hosted-url',
+        siteRoot,
+        '--link-to-remote',
+        '--no-validate-links',
+      ];
+      if (envConfig.toolEnvDartSdkDir != null) {
+        args.addAll(['--sdk-dir', envConfig.toolEnvDartSdkDir]);
+      }
+      final pr = await runProc(
+        'dart',
+        ['bin/pub_dartdoc.dart']..addAll(args),
+        workingDirectory: _pkgPubDartdocDir,
+        timeout: _dartdocTimeout * 2,
```

Comment:
Why times 2?

Reply:
Generating docs for the SDK takes longer time, and this seemed to be a good number. I'll refactor it into a separate const.